include/powerloader/download_target.hpp

```diff
@@ -36,54 +36,224 @@ namespace powerloader
         using end_callback_t = std::function<CbReturnCode(TransferStatus, const Response&)>;
         using progress_callback_t = std::function<int(curl_off_t, curl_off_t)>;
 
-        DownloadTarget(const std::string& path, const std::string& base_url, const fs::path& fn);
+        DownloadTarget(const std::string& path,
+                       const std::string& base_url,
+                       const fs::path& filename);
         ~DownloadTarget();
 
-        void set_cache_options(const CacheControl& cache_control);
-        void add_handle_options(CURLHandle& handle);
-
         DownloadTarget(const DownloadTarget&) = delete;
         DownloadTarget& operator=(const DownloadTarget&) = delete;
         DownloadTarget(DownloadTarget&&) = delete;
         DownloadTarget& operator=(DownloadTarget&&) = delete;
 
+        void set_cache_options(const CacheControl& cache_control);
+        void add_handle_options(CURLHandle& handle);
+
         bool has_complete_url() const;
         bool validate_checksum(const fs::path& path);
         bool already_downloaded();
 
         void set_error(const DownloaderError& err);
 
+        bool resume() const noexcept
+        {
+            return m_resume;
+        }
 
-        bool is_zchunk = false;
-        bool resume = true;
-        bool no_cache = false;
+        void set_resume(bool new_value) noexcept
+        {
+            m_resume = new_value;
+        }
 
-        std::string complete_url;
-        std::string path, base_url;
-        std::unique_ptr<FileIO> outfile;
+        bool no_cache() const noexcept
+        {
+            return m_no_cache;
+        }
 
-        fs::path fn;
+        const std::string& base_url() const noexcept
+        {
+            return m_base_url;
+        }
 
-        std::size_t byterange_start = 0, byterange_end = 0;
-        std::string range;
-        std::ptrdiff_t expected_size = 0;
-        std::ptrdiff_t orig_size = 0;
+        void clear_base_url()
+        {
+            m_base_url.clear();
+        }
 
-        progress_callback_t progress_callback;
-        end_callback_t end_callback;
 
-        // these are available checksums for the entire file
-        std::vector<Checksum> checksums;
+        const std::string& complete_url() const noexcept
+        {
+            return m_complete_url;
+        }
+
+        const std::string& path() const noexcept
+        {
+            return m_path;
+        }
+
+        const std::filesystem::path& filename() const noexcept
+        {
+            return m_filename;
+        }
+
+        std::size_t byterange_start() const noexcept
+        {
+            return m_byterange_start;
+        }
+
+        std::size_t byterange_end() const noexcept
+        {
+            return m_byterange_end;
+        }
+
+        std::ptrdiff_t expected_size() const noexcept
+        {
+            return m_expected_size;
+        }
+
+        // TOOD: check SOC (why is this modifed outside)
```

Comment:
With conda metadata, we know the filesize from the `repodata.json` and would like to use that for validation purposes.